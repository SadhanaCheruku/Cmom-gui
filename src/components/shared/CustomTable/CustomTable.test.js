import { render } from "@testing-library/react";
import CustomTable from "./CustomTable";
import React from "react";
import { BrowserRouter as Router } from "react-router-dom";
describe("CustomTable Repository details", () => {
  const columns = [
    { id: "name", label: "Name", minWidth: 100 },
    {
      id: "image",
      label: "Image",
      minWidth: 100,
      align: "right",
      format: (value) => value.toLocaleString("en-US"),
      renderAsImage: true,
    },
    {
      id: "symbol",
      label: "Symbol",
      minWidth: 100,
      align: "right",
      format: (value) => value,
      renderAsImage: false,
    },
    {
      id: "current_price",
      label: "Current Price",
      minWidth: 100,
      align: "right",
      format: (value) => value,
      renderAsImage: false,
    },
    {
      id: "high_24h",
      label: "High 24 hour Price",
      minWidth: 100,
      align: "right",
      format: (value) => value,
      renderAsImage: false,
    },
    {
      id: "low_24h",
      label: "Low 24 hour Price",
      minWidth: 100,
      align: "right",
      format: (value) => value,
      renderAsImage: false,
    },
  ];

  const data = [
    {
      id: 1,
      name: "BitCoin",
      image:
        "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
      symbol: "btc",
      current_price: 39310,
      high_24h: 40131,
      low_24h: 37907,
    },

    {
      id: 2,
      name: "ethereum",
      image:
        "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
      symbol: "eth",
      current_price: 39310,
      high_24h: 40131,
      low_24h: 37907,
    },
    {
      id: 3,
      name: "Tether",
      image:
        "https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579",
      symbol: "usdt",
      current_price: 39310,
      high_24h: 40131,
      low_24h: 37907,
    },
  ];
  it("render component as expected", () => {
    const { asFragment } = render(
      <Router>
        <CustomTable
          columns={columns}
          searchResult={data}
          title={"European Cryptocurrency Prices and Market Capitalization"}
        />
      </Router>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
