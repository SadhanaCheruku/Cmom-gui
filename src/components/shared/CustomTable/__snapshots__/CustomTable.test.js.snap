// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomTable Repository details render component as expected 1`] = `
<DocumentFragment>
  <div
    class="MuiGrid-root MuiGrid-container MuiGrid-direction-xs-column MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  >
    <h5
      class="MuiTypography-root makeStyles-typoGraphy-2 MuiTypography-h5 MuiTypography-colorPrimary"
    >
      European Cryptocurrency Prices and Market Capitalization
    </h5>
    <div
      class="MuiPaper-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiTableContainer-root"
      >
        <table
          aria-label="sticky table"
          class="MuiTable-root MuiTable-stickyHeader"
        >
          <thead
            class="MuiTableHead-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-head"
            >
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 100px;"
              >
                Name
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 100px;"
              >
                Image
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 100px;"
              >
                Symbol
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 100px;"
              >
                Current Price
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 100px;"
              >
                High 24 hour Price
              </th>
              <th
                class="MuiTableCell-root MuiTableCell-head MuiTableCell-alignRight MuiTableCell-stickyHeader"
                scope="col"
                style="min-width: 100px;"
              >
                Low 24 hour Price
              </th>
            </tr>
          </thead>
          <tbody
            class="MuiTableBody-root"
          >
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                BitCoin
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <img
                  alt="failed to load"
                  height="50"
                  src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
                  width="50"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                btc
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                39310
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                40131
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                37907
              </td>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                ethereum
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <img
                  alt="failed to load"
                  height="50"
                  src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
                  width="50"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                eth
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                39310
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                40131
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                37907
              </td>
            </tr>
            <tr
              class="MuiTableRow-root MuiTableRow-hover"
              role="checkbox"
              tabindex="-1"
            >
              <td
                class="MuiTableCell-root MuiTableCell-body"
              >
                Tether
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                <img
                  alt="failed to load"
                  height="50"
                  src="https://assets.coingecko.com/coins/images/1/large/bitcoin.png?1547033579"
                  width="50"
                />
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                usdt
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                39310
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                40131
              </td>
              <td
                class="MuiTableCell-root MuiTableCell-body MuiTableCell-alignRight"
              >
                37907
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <div
        class="MuiTablePagination-root"
      >
        <div
          class="MuiToolbar-root MuiToolbar-regular MuiTablePagination-toolbar MuiToolbar-gutters"
        >
          <div
            class="MuiTablePagination-spacer"
          />
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
            id="mui-23466"
          >
            Rows per page:
          </p>
          <div
            class="MuiInputBase-root MuiTablePagination-input MuiTablePagination-selectRoot"
          >
            <div
              aria-haspopup="listbox"
              aria-labelledby="mui-23466 mui-31172"
              class="MuiSelect-root MuiSelect-select MuiTablePagination-select MuiSelect-selectMenu MuiInputBase-input"
              id="mui-31172"
              role="button"
              tabindex="0"
            >
              10
            </div>
            <input
              aria-hidden="true"
              class="MuiSelect-nativeInput"
              tabindex="-1"
              value="10"
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSelect-icon MuiTablePagination-selectIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M7 10l5 5 5-5z"
              />
            </svg>
          </div>
          <p
            class="MuiTypography-root MuiTablePagination-caption MuiTypography-body2 MuiTypography-colorInherit"
          >
            1-3 of 3
          </p>
          <div
            class="MuiTablePagination-actions"
          >
            <button
              aria-label="Previous page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Previous page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.41 16.09l-4.58-4.59 4.58-4.59L14 5.5l-6 6 6 6z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Next page"
              class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit Mui-disabled Mui-disabled"
              disabled=""
              tabindex="-1"
              title="Next page"
              type="button"
            >
              <span
                class="MuiIconButton-label"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M8.59 16.34l4.58-4.59-4.58-4.59L10 5.75l6 6-6 6z"
                  />
                </svg>
              </span>
            </button>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
